core/gui/src/app/workspace/service/report-generation/report-generation.service.ts

```diff
@@ -67,27 +69,236 @@ export class ReportGenerationService {
   }
 
   /**
-   * Generates an HTML file containing the workflow snapshot and triggers a download of the file.
-   * @param {string} workflowSnapshotURL - The base64-encoded PNG image URL of the workflow snapshot.
-   * @param {string} workflowName - The name of the workflow.
+   * Retrieves and processes results for all specified operators within the workflow.
+   * This function iterates over each operator ID, fetches the corresponding result details via `retrieveOperatorInfoReport`,
+   * and collects these results into an array. The function returns a promise that resolves with the processed results,
+   * which can be used to generate a comprehensive HTML report or for further processing.
+   *
+   * @param {Object} payload - An object containing the operator IDs that need to be processed.
+   * @param {string[]} payload.operatorId - An array of operator IDs representing each operator in the workflow.
+   *
+   * @returns {Promise<{operatorId: string, html: string}[]>} - A promise that resolves with an array of objects,
+   * each containing an `operatorId` and its corresponding HTML representation of the result.
+   * This result array can be used to generate an HTML report or for other purposes.
+   */
+
+  public getAllOperatorResults(payload: { operatorId: string[] }): Promise<{ operatorId: string; html: string }[]> {
```

Comment:
Can you use Observe<{operatorId: string; operatorResultHtml: string}[]> as the return parameter type and adjust the methods.